src/libYARP_dev/src/devices/RobotDescriptionClient/RobotDescriptionClient.cpp

```diff
@@ -0,0 +1,182 @@
+
+    b.addVocab(VOCAB_IROBOT_DESCRIPTION);
+    b.addVocab(VOCAB_IROBOT_GET);
+    b.addVocab(VOCAB_IROBOT_CONTROLBOARD_WRAPPERS);
+    bool ret = m_rpc_port.write(b, resp);
+    if (ret)
+    {
+        if (resp.get(0).asVocab() != VOCAB_OK)
+        {
+            yError() << "RobotDescriptionClient::getAllDevices() received error from server";
+            return false;
+        }
+        else
+        {
+            Bottle *b = resp.get(1).asList();
+            for (int i = 0; i < b->size(); i += 2)
+            {
+                RobotDescription desc;
+                desc.device_name = b->get(i).asString();
+                desc.device_type = b->get(i + 1).asString();
+                dev_list.push_back(desc);
+            }
+            return true;
+        }
+    }
+    else
+    {
+        yError() << "RobotDescriptionClient: error on writing on rpc port";
+        return false;
+    }
+    return true;
+}
+
+bool yarp::dev::RobotDescriptionClient::registerDevice(const RobotDescription& dev)
+{
+    yarp::os::Bottle b;
+    yarp::os::Bottle resp;
+
+    b.addVocab(VOCAB_IROBOT_DESCRIPTION);
+    b.addVocab(VOCAB_IROBOT_SET);
+    b.addVocab(VOCAB_IROBOT_DEVICE);
+    b.addString(dev.device_name);
+    b.addString(dev.device_type);
+    bool ret = m_rpc_port.write(b, resp);
+    if (ret)
+    {
+        if (resp.get(0).asVocab() != VOCAB_OK)
+        {
+            yError() << "RobotDescriptionClient::registerDevice() received error from server";
+            return false;
+        }
+    }
+    else
+    {
+        yError() << "RobotDescriptionClient: error on writing on rpc port";
+        return false;
+    }
+    return true;
+}
+
+bool yarp::dev::RobotDescriptionClient::getAllDevices(std::vector<RobotDescription>& dev_list)
+{
+    yarp::os::Bottle b;
+    yarp::os::Bottle resp;
+
+    b.addVocab(VOCAB_IROBOT_DESCRIPTION);
+    b.addVocab(VOCAB_IROBOT_GET);
+    b.addVocab(VOCAB_IROBOT_ALL);
+    bool ret = m_rpc_port.write(b, resp);
+    if (ret)
+    {
+        if (resp.get(0).asVocab() != VOCAB_OK)
+        {
+            yError() << "RobotDescriptionClient::getAllDevices() received error from server";
+            return false;
+        }
+        else
+        {
+            Bottle *b = resp.get(1).asList();
+            for (int i = 0; i < b->size();i+=2) 
+            {
+                RobotDescription desc;
+                desc.device_name = b->get(i).asString();
+                desc.device_type = b->get(i+1).asString();
+                dev_list.push_back(desc);
+            }
+            return true;
+        }
+    }
+    else
+    {
+        yError() << "RobotDescriptionClient: error on writing on rpc port";
+        return false;
+    }
+    return true;
+}
+
+yarp::dev::DriverCreator *createRobotDescriptionClient()
+{
+    return new DriverCreatorOf<RobotDescriptionClient>
+               (
+                   "robotDescriptionClient",
+                   "",
+                   "robotDescriptionClient"
```

Comment:
This should be `RobotDescriptionClient`, as from the documentation of this constructor [1] this parameter is "The name of a class associated with this device. This is to give the user an entry point into the relevant code documentation." .

[1] : http://www.yarp.it/classyarp_1_1dev_1_1DriverCreatorOf.html#a0918fdc89443a9e6217bd02af78c0db6